vminitd/Sources/vmexec/RunCommand.swift

```diff
@@ -104,11 +132,17 @@ struct RunCommand: ParsableCommand {
                 }
             }
 
+            var fdCopy = hostFd
+            var fdData = Data(bytes: &fdCopy, count: MemoryLayout.size(ofValue: fdCopy))
+            try childPipe.fileHandleForWriting.write(contentsOf: fdData)
+            try childPipe.fileHandleForWriting.close()
+
             // Apply O_CLOEXEC to all file descriptors except stdio.
             // This ensures that all unwanted fds we may have accidentally
             // inherited are marked close-on-exec so they stay out of the
             // container.
-            try App.applyCloseExecOnFDs()
+            let preserve: Set<Int32> = hostFd >= 0 ? [hostFd] : []
+            try App.applyCloseExecOnFDs(preserve: preserve)
```

Comment:
Same comment as in ExecCommand, we'll want to close this, but have some way for the parent to ack that it received the fd first and it's all good to start receiving and sending io.